cmd/snap-confine/snap-device-helper

```diff
@@ -53,8 +53,20 @@ if [ ! -e "$app_dev_cgroup" ]; then
 fi
 
 # check if it's a block or char dev
-if [ "${DEVPATH#*/block/}" != "$DEVPATH" ]; then
+# TODO: re-write this to be more robust, the bash variable substitution done 
+# here is quite awkard :-/
+if [ "${DEVPATH#*/block/}" != "$DEVPATH" ] ; then
     type="b"
+elif [ "${DEVPATH#*/nvme/}" != "$DEVPATH" ]; then
+    # nvme subsystem devices can either be block or character devices, the
+    # block devices have a more deeply nested path than the character devices
+    NVME_SUBPATH="${DEVPATH#*/nvme*/}"
+    if [ "${NVME_SUBPATH#*/}" != "$NVME_SUBPATH" ]; then
+        # it is more deeply nested
+        type="b"
+    else
+        type="c"
+    fi
 else
     type="c"
 fi
```

Comment:
You're right my initial suggestion wasn't quite right. I think this is a little simpler and achieves the same:
```
if [ "${DEVPATH#*/block/}" != "$DEVPATH" ]; then
    type="b"
elif [ "${DEVPATH#*/nvme/nvme*/nvme*n*}" != "$DEVPATH" ]; then
    # char devices are .../nvme/nvme* but block devices are
    # .../nvme/nvme*/nvme*n* and .../nvme/nvme*/nvme*n*p*
    # so if have a device that has nvme/nvme*/nvme*n* in it,
    # treat it as a block device
    type="b"
else
    type="c"
fi
```